model/Image.php

```diff
@@ -697,12 +697,25 @@ public function getFormattedImage($format) {
 	public function cacheFilename($format) {
 		$args = func_get_args();
 		array_shift($args);
+		
+		// Note: $folder holds the *original* file, while the Image we're working with
+		// may be a formatted image in a child directory (this happens when we're chaining formats)
 		$folder = $this->ParentID ? $this->Parent()->Filename : ASSETS_DIR . "/";
-
+		
 		$format = $format . Convert::base64url_encode($args);
-		$filename = $format . "-" . $this->Name;
-		$patterns = $this->getFilenamePatterns($this->Name);
-		if (!preg_match($patterns['FullPattern'], $filename)) {
+		$filename = $format . "/" . $this->Name;
+		
+		$pattern = $this->getFilenamePatterns($this->Name);
+		
+		// Any previous formats need to be derived from this Image's directory, and prepended to the new filename
+		$prepend = array();
```

Comment:
So each manipulation on a file is a new subdirectory?

E.g. `$Image.SetWidth(100).SetHeight(200)`?